### PR TITLE
MGMT-9747: adding image validation regex to steps  request objects

### PR DESCRIPTION
**Pull request by tsorya**
[MGMT-9747](https://issues.redhat.com/browse/MGMT-9747): adding image validation regex to steps  request objects
[MGMT-9893](https://issues.redhat.com/browse/MGMT-9893): Add validation for domain name resolution and host nic

## List all the issues related to this PR

- [x] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @avishayt 
/cc @eranco74 

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: swagger.yaml [hunk 1]
```diff
@@ -5692,9 +5696,11 @@ definitions:
         type: string
         description: Boot device to write image on
       controller_image:
+        pattern: '^(([a-zA-Z0-9\-\.]+)(:[0-9]+)\/)?[a-z0-9\._\-\/@]+:[a-zA-Z0-9_\-.]+$'
```
Comment (ori-amizur) [inline swagger.yaml:5699]: Seems that this does not allow image without tag, although it should be valid.

@avishayt what do you think ?

Comment (avishayt) [reply swagger.yaml:5699]: The rule of thumb should be that we allow all legal configurations (so we don't shoot ourselves in the foot) while blocking illegal configurations (for security reasons).

**Review by eranco74 — COMMENTED** [review]
/lgtm

**Pull request merged**